### PR TITLE
Threshold, clip and RGB normalize fixed

### LINKED ISSUE #180: Update dependencies
**Is your feature request related to a problem? Please describe.**
Currently, we are dependent on significantly old versions of TorchIO and SimpleITK, and we should move to the latest versions for added functionality and optimization.

**Describe the solution you'd like**
N.A.

**Describe alternatives you've considered**
N.A.

**Additional context**
N.A.

### LINKED ISSUE #189: Threshold/clip get applied to label images as well
**Describe the bug**
Threshold and clip operations are getting applied to label files, which should not be the case.

**To Reproduce**
1. Add `'clip'` or `'threshold'` in `data_preprocessing` of config
2. Ensure there is a label file present in data
3. See this getting applied to the label file along with images

**Expected behavior**
These operations should be for the intensity images only, like normalization.

**GaNDLF Version**
0.0.12-dev

**Additional context**
This would probably also happen for `normalize_rgb` submodule.

### PR DESCRIPTION
**Pull request by sarthakpati**
Fixes #189 
Fixes #180 

## Proposed Changes

- previous implementation worked on both intensity and label images, which we don't want

## Checklist

- [x] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [x] My PR is based from the [current GaNDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [x] Non-breaking change (would not break existing functionality): if changes breaks current code, please provide as many details as possible
- [x] Function/class source code documentation added/updated
- [x] Code has been [blacked](https://github.com/psf/black#usage) for style consistency
- [x] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/version.py)
- [x] If adding a submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/CBICA/GaNDLF/blob/master/pyproject.toml) file 
- [x] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [x] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [x] Tests added or modified to [cover the changes](https://app.codecov.io/gh/CBICA/GaNDLF); if coverage is reduced, please give explanation

### CODE REVIEW

File: GANDLF/data/preprocessing/threshold_and_clip.py [hunk 1]
```diff
@@ -1,45 +1,64 @@
...
+            self.apply_threshold(image)
+        return subject
+
+    def apply_threshold(self, image: ScalarImage) -> None:
+        image.set_data(self.threshold(image.data))
+
+    def threshold(self, tensor: torch.Tensor) -> torch.Tensor:
+        C = torch.zeros(tensor.shape, dtype=tensor.dtype)
+        l1_tensor = torch.where(tensor < self.out_max, tensor, C)
+        output = torch.where(l1_tensor > self.out_min, l1_tensor, C)
```
Comment (Geeks-Sid) [inline GANDLF/data/preprocessing/threshold_and_clip.py:54]: Not sure whether this should happen.

Comment (sarthakpati) [reply GANDLF/data/preprocessing/threshold_and_clip.py:54]: How else to do threshold?

Comment (Geeks-Sid) [reply GANDLF/data/preprocessing/threshold_and_clip.py:54]: `l1_tensor = torch.where(tensor < self.out_max, tensor, self.out_max)
output = torch.where(l1_tensor > self.out_min, l1_tensor, self.out_min)`

This should be the way, no?

Comment (sarthakpati) [reply GANDLF/data/preprocessing/threshold_and_clip.py:54]: That is the logic for clip/clamp. This is threshold.

```
if (x < min_thresh) or (x > max_thresh):
    x = 0
```

**Review by Geeks-Sid — APPROVED** [review]
